app/utils.js

```diff
@@ -35,47 +35,39 @@ function notify(str) {
   */
 }
 
-function gcmCompliant() {
+function loadShim(polyfill) {
+  return new Promise((resolve, reject) => {
+    const shim = document.createElement('script');
+    shim.src = polyfill;
+    shim.addEventListener('load', () => resolve(true));
+    shim.addEventListener('error', () => resolve(false));
```

Comment:
I can't remember how these are being used, but should we be doing `resolve(false)` or just rejecting the promise?

Reply:
It's just a design choice. I'd rather return a boolean than use a try/catch block on the await.